sktime/transformers/rocket.py

```diff
@@ -0,0 +1,178 @@
+from numba import njit, prange
+import numpy as np
+import pandas as pd
+
+from .base import BaseTransformer
+
+__author__ = "Angus Dempster"
+__all__ = ["ROCKET"]
+
+class Rocket(BaseTransformer):
+    """ROCKET
+
+    RandOm Convolutional KErnel Transform
+
+    @article{dempster_etal_2019,
+      author  = {Dempster, Angus and Petitjean, Fran\c{c}ois and Webb, Geoffrey I},
+      title   = {ROCKET: Exceptionally fast and accurate time series classification using random convolutional kernels},
+      year    = {2019},
+      journal = {arXiv:1910.13051}
+    }
+
+    Parameters
+    ----------
+    input_length : int, length of input time series
+    num_kernels  : int, number of random convolutional kernels
+    num_channels : int, number of channels for multivariate time series (default 1)
```

Comment:
What do you mean by channels? Variables in the multivariate setting?

Reply:
I think `channel == dimension`, so each column in a sktime pandas DataFrame would be a channel in this terminology (coming from cnns).  Happy to change.

Reply:
Alright, we've started writing a little [glossary](https://github.com/alan-turing-institute/sktime/wiki/Glossary) to keep terminology consistent: so variables/channels/dimensions are referred to as columns.